### PR TITLE
Update instructions for auth api tests.

### PR DESCRIPTION
**Pull request by rsgowman**

### CODE REVIEW

**Comment by rsgowman** [comment]
I initially struggled to get the apitests to work. The changes in this PR allowed them to pass, though that doesn't necessarily mean I've done the right thing, so a sanity check on what I'm doing (in addition to the mechanics) would be appreciated!

File: Example/Auth/ApiTests/GoogleAuthTests.m [hunk 1]
```diff
@@ -63,12 +63,16 @@ - (void)testSignInWithGoogle {
                                            error.localizedDescription);
                                  }
                                }];
-  XCTAssertEqualObjects(auth.currentUser.displayName, kGoogleTestAccountName);
+
+  if (auth.currentUser.displayName) {
```
Comment (rsgowman) [inline Example/Auth/ApiTests/GoogleAuthTests.m]: See reference below (line 74-75). It seems to be an issue in practice; one of my test accounts did not have an id_token returned by the refresh flow.

File: Example/Auth/ApiTests/AuthCredentialsTemplate.h [hunk 2]
```diff
@@ -62,7 +72,10 @@ The ID of the test user in the Custom Auth token.
...
-#define KCUSTOM_AUTH_TOKEN_URL $KCUSTOM_AUTH_TOKEN_URL
-#define KCUSTOM_AUTH_TOKEN_EXPIRED_URL $KCUSTOM_AUTH_TOKEN_EXPIRED_URL
-#define KCUSTOM_AUTH_USER_ID $KCUSTOM_AUTH_USER_ID
+#define KFACEBOOK_USER_NAME @"Fname Lname"
+
+// The custom auth test is currently disabled. These values are ignored (though
+// must be present).
+#define KCUSTOM_AUTH_TOKEN_URL @"Disabled test"
+#define KCUSTOM_AUTH_TOKEN_EXPIRED_URL @"Disabled test"
+#define KCUSTOM_AUTH_USER_ID @"Disabled test"
```
Comment (renkelvin) [inline Example/Auth/ApiTests/AuthCredentialsTemplate.h]: I think it's better to have macros in the Template files or developers may think `KFACEBOOK_USER_NAME` has to be `Fname Lname`

File: Example/Auth/README.md [hunk 3]
```diff
@@ -13,21 +13,40 @@ identifier (e.g. `com.google.FirebaseExperimental1.dev`)
 4. Download the resulting `GoogleService-Info.plist` and place it in
 [Sample/GoogleService-Info.plist](Sample/GoogleService-Info.plist)
 
-#### GoogleService-Info_multi.plist files
+#### GoogleService-Info\_multi.plist files
```
Comment (renkelvin) [inline Example/Auth/README.md]: I don't really understand this

File: Example/Auth/README.md [hunk 4]
```diff
@@ -84,6 +103,18 @@ Please follow the instructions in
...
+
+In the Firebase conosle for your test project, you'll need to enable the
+following auth providers:
+* Email/Password
+* Google
+* Facebook
+* Anonymous
+
+You'll also need to create a user with email
+`iosgcip+email_existing_user@gmail.com` and password of `password`.
```
Comment (renkelvin) [inline Example/Auth/README.md]: Our testing code was not designed for public developers. We should use some place holder here.

**Comment by ryanwilson** [comment]
@renkelvin is this something you want to get landed, or pieces of it like documentation improvements?

**Comment by renkelvin** [comment]
@ryanwilson Yeah, it's a pretty valuable improvement to the documentation.
@rsgowman Could you please take a look at the open issues when you get some time? Thanks!

**Comment by renkelvin** [comment]
As for the api tests for custom auth, I think we can change it back when reenable the tests.

BTW, it seems there is some issue with CI. I'm not sure what the root cause is, but last time it's solved after I created a new pr.

**Pull request merged**